controllers/account/account_controller.go

```diff
@@ -367,6 +410,12 @@ func (r *AccountReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 
 		}
 
+		optInRegions, ok := configMap.Data["opt-in-regions"]
+		// handles region enablement for supported Opt-In regions
+		if (ok && optInRegions != "" && isEnabled && !currentAcctInstance.Spec.BYOC) && (currentAcctInstance.Status.State == AccountCreating || currentAcctInstance.Status.State == AccountOptingInRegions) {
+			return r.handleOptInRegionEnablement(reqLogger, currentAcctInstance, awsSetupClient, optInRegions, numberOfAccountsOptingIn)
+		}
+
```

Comment:
I think we need to rethink how we get to this point - I think at the end of the AccountCreating step we should decide what the next state is and put the CR into that state directly, and then return the reconciler - then when it gets back to this point in the code it skips the Account Creation stuff and gets here and goes "oh, I'm AccountOptingInRegions - I'll do that handler" and returns from that, then when we're done handling the opt-in region enablement we tell that to put it to the next state, instead of relying on it to just fall through as we're doing here for the InitializeRegions.

If that doesn't make sense, let me know I'll be happy to explain.